includes/core/class-wc-payments-customer-service-api.php

```diff
@@ -0,0 +1,183 @@
+	 *
+	 * @return string The created customer's ID
+	 *
+	 * @throws API_Exception Error creating customer.
+	 */
+	public function create_customer_for_user( WP_User $user, array $customer_data ): string {
+		return $this->customer_service->create_customer_for_user( $user, $customer_data );
+	}
+
+	/**
+	 * Update the customer details held on the WCPay server associated with the given WordPress user.
+	 *
+	 * @param string  $customer_id WCPay customer ID.
+	 * @param WP_User $user        WordPress user.
+	 * @param array   $customer_data Customer data.
+	 *
+	 * @return string The updated customer's ID. Can be different to the ID parameter if the customer was re-created.
+	 *
+	 * @throws API_Exception Error updating the customer.
+	 */
+	public function update_customer_for_user( string $customer_id, WP_User $user, array $customer_data ): string {
+		return $this->customer_service->update_customer_for_user( $customer_id, $user, $customer_data );
+	}
+
+	/**
+	 * Sets a payment method as default for a customer.
+	 *
+	 * @param string $customer_id       The customer ID.
+	 * @param string $payment_method_id The payment method ID.
+	 */
+	public function set_default_payment_method_for_customer( $customer_id, $payment_method_id ) {
+		return $this->customer_service->set_default_payment_method_for_customer( $customer_id, $payment_method_id );
+	}
+
+	/**
+	 * Gets all payment methods for a customer.
+	 *
+	 * @param string $customer_id The customer ID.
+	 * @param string $type        Type of payment methods to fetch.
+	 *
+	 * @throws API_Exception We only handle 'resource_missing' code types and rethrow anything else.
+	 */
+	public function get_payment_methods_for_customer( $customer_id, $type = 'card' ) {
+		return $this->customer_service->get_payment_methods_for_customer( $customer_id, $type );
+	}
+
+	/**
+	 * Updates a customer payment method.
+	 *
+	 * @param string   $payment_method_id The payment method ID.
+	 * @param WC_Order $order             Order to be used on the update.
+	 */
+	public function update_payment_method_with_billing_details_from_order( $payment_method_id, $order ) {
+		return $this->customer_service->update_payment_method_with_billing_details_from_order( $payment_method_id, $order );
+	}
+
+	/**
+	 * Clear payment methods cache for a user.
+	 *
+	 * @param int $user_id WC user ID.
+	 */
+	public function clear_cached_payment_methods_for_user( $user_id ) {
+		return $this->customer_service->clear_cached_payment_methods_for_user( $user_id );
+	}
+
+	/**
+	 * Given a WC_Order or WC_Customer, returns an array representing a Stripe customer object.
+	 * At least one parameter has to not be null.
+	 *
+	 * @param WC_Order    $wc_order    The Woo order to parse.
+	 * @param WC_Customer $wc_customer The Woo customer to parse.
+	 *
+	 * @return array Customer data.
+	 */
+	public static function map_customer_data( WC_Order $wc_order = null, WC_Customer $wc_customer = null ): array {
```

Comment:
I really like type-hinting, but the mos of the codebase doesn't have it. Shall we maybe consider removing function type-hinting?

Reply:
Its present in `Customer_Service` class so I think it would be better to retain it here.

Reply:
Oh, that's nice. We can leave it. I was afraid that you could get other variation of order (overridden one we have in our  codebase), but if it's already there, we can leave it.